backend/src/main/java/com/redhat/cloud/notifications/db/repositories/NotificationRepository.java

```diff
@@ -23,22 +24,39 @@ public class NotificationRepository {
     @Inject
     EntityManager entityManager;
 
-    public List<NotificationHistory> getNotificationHistory(String tenant, UUID endpoint, boolean includeDetails, Query limiter) {
+    @Inject
+    OrgIdHelper orgIdHelper;
+
+    public List<NotificationHistory> getNotificationHistory(String tenant, String orgId, UUID endpoint, boolean includeDetails, Query limiter) {
         String query = "SELECT NEW NotificationHistory(nh.id, nh.invocationTime, nh.invocationResult, nh.endpoint, nh.created";
         if (includeDetails) {
             query += ", nh.details";
         }
-        query += ") FROM NotificationHistory nh WHERE nh.event.accountId = :accountId AND nh.endpoint.id = :endpointId";
+        query += ") FROM NotificationHistory nh WHERE nh.endpoint.id = :endpointId AND ";
+        if (orgIdHelper.useOrgId(orgId)) {
```

Comment:
Is there any specific reason the `orgIdHelper` is used here but the `FeatureFlipper` elsewhere?

Reply:
`FeatureFlipper` is used in places where the `accountId` can be `null`, like methods dealing with default behavior groups for example. `OrgIdHelper` is used everywhere else.